src/fs/filesystem.ts

```diff
@@ -336,24 +395,36 @@ export class FileSystem {
       throw new Error("Destination already exists")
     }
 
-    await this.runOnNode(from, true, (node, relPath) => {
-      if (typeCheck.isFile(node)) {
-        throw new Error("Tried to `mv` within a file")
-      }
+    await this.runOnNode(from, {
+      public: async (root, relPath) => {
+        const [_, ...nextPath] = pathing.unwrap(to)
+        await root.mv(relPath, nextPath)
+      },
+      private: async (node, relPath) => {
+        if (typeCheck.isFile(node)) {
+          throw new Error("Tried to `mv` within a file")
+        }
 
-      const [ _, ...nextPath ] = pathing.unwrap(to)
-      return node.mv(relPath, nextPath)
+        const [_, ...nextPath] = pathing.unwrap(to)
+        // TODO FIXME: nextPath is wrong if you use a node that's deeper in the tree.
```

Comment:
Fix for after this PR?

Reply:
Yeah, IIRC, when I was going though the code I noticed this was probably the source of this bug: https://github.com/fission-codes/webnative/issues/238

I can try that in another PR after this :)